app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -0,0 +1,129 @@
+/** @file Provides {@link Cognito} class which is the entrypoint into the AWS Amplify library.
+ *
+ * All of the functions used for authentication are provided by the AWS Amplify library, but we
+ * provide a thin wrapper around them to make them easier to use. Mainly, we perform some error
+ * handling and conditional logic to vary behaviour between desktop & cloud. */
+import * as amplify from '@aws-amplify/auth'
+import * as cognito from 'amazon-cognito-identity-js'
+import * as results from 'ts-results'
```

Comment:
are `results` a lot better than (say) returning `| Error`?

Reply:
If we're just talking about returning an ok and an error response then no, it's the same. But this offers additional benefits like composing operations with `map` and `mapErr` so I felt like we should use it when building our API class.